models/fivetran_log__audit_table.sql

```diff
@@ -1,3 +1,19 @@
+{{ config(
+    materialized='incremental',
+    partition_by={
+        'field': 'sync_start',
+        'data_type': 'timestamp',
+        'granularity': 'day'
+    }
+) }}
+
+{%- call statement('max_sync_start', fetch_result=True) -%}
+    select date(max(sync_start)) from {{ this }}
+{%- endcall -%}
+
+{%- set query = load_result('max_sync_start') -%}
+{%- set max_sync_start = query['data'][0][0] -%}
+
```

Comment:
@DylanBaker i've never seen this way to limit to the new increments -- is this (having a call statement, referring to that query output in the `is_incremental()` block) more performant or anything compared to having a subquery straight in the `is_incremental` block? or more of a stylistic choice?

Reply:
On everything that isn't BigQuery, it's a stylistic choice (and possibly even a less performant one, but on BigQuery a subquery doesn't cause a partition to get pruned correctly, so you don't benefit at all from the incremental logic from a query cost point of view. (i.e. it still scans the whole table even if the subquery would tell you to do otherwise.) It needs a literal value. It's very dumb.

Reply:
oh wow i didn't know that! so we should probably be employing this kind of incremental logic across the board the board ...